catalyst_voices/packages/internal/catalyst_voices_models/lib/src/document/document_ref.dart

```diff
@@ -72,6 +72,34 @@ sealed class DocumentRef extends Equatable implements Comparable<DocumentRef> {
     return 0;
   }
 
+  /// Checks if this reference encompasses the [other] reference.
+  ///
+  /// Returns `true` if:
+  /// - Both references have the same runtime type (e.g. both are [DraftRef]).
+  /// - Both references have the same [id].
+  /// - This reference includes the version of [other].
+  ///
+  /// Specific behavior regarding versions:
+  /// - If this reference is **loose** ([ver] is `null`), it contains any version of [other]
+  ///   (both loose and exact).
+  /// - If this reference is **exact** ([ver] is not `null`), it only contains [other] if
+  ///   [other] has the exactly same [ver].
+  bool contains(DocumentRef other) {
+    if (runtimeType != other.runtimeType) return false;
+
+    if (id != other.id) return false;
+
+    final ver = this.ver;
+    final otherVer = other.ver;
+
+    // contains all versions
+    if (ver == null) return true;
+    // only specific version
+    if (ver != otherVer) return false;
+
+    return true;
```

Comment:
This could be written as such too. Maybe you'll find it more readable, for me it is. 
```suggestion
    return ver == null || ver == otherVer;
```

This part:
```dart
if (ver != otherVer) return false;
```
makes me stop for a longer moment because it uses negation instead of direct comparison.